src/components/pagination/Pagination.tsx

```diff
@@ -1,120 +1,60 @@
-
-  const pagesArray = Array.from({ length: pages }, (_, index) => (1 + index).toString());
-
-  const [currentBlock, setCurrentBlock] = useState(firstBlock);
-
-  const isFirstBlock = currentBlock === firstBlock;
-  const isLastBlock = currentBlock === lastBlock;
-
-  const blockFirstPage = currentBlock * 4 + 2;
-  const blockLastPage = (currentBlock + 1) * 4;
-
-  const isBlockFirstPage = +currentPage === blockFirstPage;
-  const isBlockLastPage = +currentPage === blockLastPage;
-
-  const list = useRef<HTMLUListElement>(null);
-
-  useEffect(() => {
-    if (currentPage === '1') {
-      setCurrentBlock(firstBlock);
-    }
-
-    if (currentPage === pages.toString()) {
-      setCurrentBlock(lastBlock);
-    }
-  }, [currentPage, lastBlock, pages]);
-
-  const onLeftMove = () => {
-    if (currentPage !== '1') {
-      setPage((+currentPage - 1).toString());
-      if (+currentPage <= blockFirstPage) {
-        setCurrentBlock(currentBlock - 1);
-      }
+import { PaginationProps } from '../../types/PaginationProps';
+import { usePagination } from '../../hooks/UsePagination';
+
+export const Pagination: React.FC<PaginationProps> = ({
+  totalCount,
+  pageSize,
+  currentPage,
+  onPageChange,
+}) => {
+  const paginationRange = usePagination({ totalCount, pageSize, currentPage });
+
+  if (paginationRange.length < 2) {
+    return null;
+  }
+
+  const onNext = () => {
+    if (currentPage < paginationRange.length) {
+      onPageChange(currentPage + 1);
     }
   };
 
-  const onRightMove = () => {
-    if (!(isBlockLastPage && isLastBlock)) {
-      setPage((+currentPage + 1).toString());
-      if (+currentPage >= blockLastPage) {
-        setCurrentBlock(currentBlock + 1);
-      }
+  const onPrevious = () => {
+    if (currentPage > 1) {
+      onPageChange(currentPage - 1);
     }
   };
 
   return (
     <div className="pagination">
-      <button className="pagination__left-slide" onClick={onLeftMove}>
-        <div
-          className={
-            isFirstBlock && isBlockFirstPage ? 'icon icon-left-disabled' : 'icon icon-left'
-          }
-        />
-      </button>
+      <div
+        onClick={onPrevious}
+        className={`pagination__button ${currentPage === 1 ? 'pagination__button--disabled' : ''}`}
+      >
+        <img className="pagination__button__previous" src={arrowLeft} alt="previous-button" />
+      </div>
 
-      <div className="pagination__slider">
-        <li className="pagination__slider-item" key={1}>
-          <button
-            className={classNames('pagination__slider-num', {
-              'pagination__slider-num-active': currentPage === '1',
-            })}
-            onClick={() => setPage('1')}
+      <ul className="numbers pagination__numbers">
+        {paginationRange.map(pageNumber => (
+          <li
+            key={pageNumber}
+            className={`numbers__page ${currentPage === pageNumber ? 'numbers__page--active' : ''}`}
+            onClick={() => onPageChange(pageNumber)}
```

Comment:
Shouldn't we have a handlePageClick to deal with this? Something like:
`const handlePageClick = (pageNumber: number) => {
  if (pageNumber !== currentPage) {
    onPageChange(pageNumber);
  }
};`